src/utils/data-access.js

```diff
@@ -35,3 +35,56 @@ export async function retrieveSiteBySiteId(dataAccess, siteId, log) {
     throw new Error(`Error getting site ${siteId}: ${e.message}`);
   }
 }
+
+/**
+ * Synchronizes existing suggestions with new data by removing outdated suggestions
+ * and adding new ones.
+ *
+ * @param {Object} params - The parameters for the sync operation.
+ * @param {Object} params.opportunity - The opportunity object to synchronize suggestions for.
+ * @param {Array} params.newData - Array of new data objects to sync.
+ * @param {Function} params.buildKey - Function to generate a unique key for each item.
+ * @param {Function} params.mapNewSuggestion - Function to map new data to suggestion objects.
+ * @param {Object} params.log - Logger object for error reporting.
+ * @returns {Promise<void>} - Resolves when the synchronization is complete.
+ */
+export async function syncSuggestions({
+  opportunity,
+  newData,
+  buildKey,
+  mapNewSuggestion,
+  log,
+}) {
+  const newDataKeys = new Set(newData.map(buildKey));
+  const existingSuggestions = await opportunity.getSuggestions();
+
+  // Remove outdated suggestions
+  await Promise.all(
+    existingSuggestions
+      .filter((existing) => !newDataKeys.has(buildKey(existing)))
+      .map((suggestion) => suggestion.remove()),
+  );
+
+  // Prepare new suggestions
+  const newSuggestions = newData
+    .filter((data) => !existingSuggestions.some(
+      (existing) => buildKey(existing) === buildKey(data),
+    ))
+    .map(mapNewSuggestion);
+
+  // Add new suggestions if any
```

Comment:
We should also update existing suggestions which are not created or modified via Back-Office-UI and but have surfaced in the new audits also.

e.g. 
If following suggestion exists:
```
{
  "url_to": "https://www.lovesac.com/terms_and_conditions",
  "title": "Lovesac returns to its 'Roots' for 20th anniversary",
  "url_from": "https://www.prnewswire.com/news-releases/lovesac-returns-to-its-roots-for-20th-anniversary-300121114.html",
  "traffic_domain": 947754
}
```
and in the new audit, we detect the same broken link again in the same referring page, but the Title or traffic numbers for that referring have changed now, we should update those in the suggestions entry.